cloudbuild.yaml

```diff
@@ -44,6 +52,28 @@ steps:
       terraform validate -no-color
     dir: 'applications/rag/'
     waitFor: ['validate platform']
+  
+  - id: 'create network'
+    name: 'gcr.io/$PROJECT_ID/terraform'
+    env:
+      - "KUBE_LOAD_CONFIG_FILE=false"
+    entrypoint: 'sh'
+    args:
+      - '-c'
+      - |
+        set -e
+
+        cd /workspace/modules/custom-network
+        terraform apply \
```

Comment:
yes you are right, combined it into cluster creation

Reply:
a I might misunderstood it so want to confirm that 
if we want to accelerate the deployment by paralleling cloudsql database and gke cluster creation, we have to move network creation out of `infrastrcuture/`. 
As cloudsql database depends on the network. 
@imreddy13

Reply:
@andrewsykim this PR splits out the network creation from infra, because for RAG we need to create the cloud SQL instance in parallel to the GKE cluster (which needs a network to exist).

But for ray and jupyter, we don't need that. @yiyinglovecoding can we test both flows in our E2E? 
1) network created by infra for ray and jupyter applications
2) network created by rag for rag application

Might need to restructure and/or add tests

Reply:
it will be around at least 10mins for AP.